charms/kfp-persistence/tests/unit/test_operator.py

```diff
@@ -1,262 +1,170 @@
+
+def test_kfp_api_relation_without_relation(harness):
+    """Test that the kfp-api relation goes Blocked if no relation is established."""
+    # Arrange
     harness.begin()
 
-    other_app = "other-app"
-    other_unit = f"{other_app}/0"
+    # Mock:
+    # * leadership_gate_component_item to be active and executed
+    harness.charm.leadership_gate.executed = True
+    harness.charm.leadership_gate.get_status = MagicMock(return_value=ActiveStatus())
 
-    if relation_data is not None:
-        rel_id = harness.add_relation(relation_name, other_app)
-        harness.add_relation_unit(rel_id, other_unit)
-        harness.update_relation_data(rel_id, other_app, relation_data)
-
-    with expected_raises as partial_relation_data:
-        interfaces = harness.charm._get_interfaces()
-        data = harness.charm._validate_sdi_interface(interfaces, relation_name)
-    if expected_status is None:
-        assert data == expected_returned_data
-    else:
-        assert partial_relation_data.value.status == expected_status
-
-
-def test_install_with_all_inputs(harness, oci_resource_data):
-    harness.set_leader()
-    model_name = "test_model"
-    harness.set_model_name(model_name)
-
-    # Set up required relations
-    # Future: convert these to fixtures and share with the tests above
-    harness.add_oci_resource(**oci_resource_data)
-
-    # kfp-api relation data
-    os_data = {
-        "_supported_versions": "- v1",
-        "data": yaml.dump(
-            {
-                "service-name": "some-service.some-namespace",
-                "service-port": "1928",
-            }
-        ),
-    }
-    os_rel_id = harness.add_relation("kfp-api", "kfp-api-provider")
-    harness.add_relation_unit(os_rel_id, "kfp-api-provider/0")
-    harness.update_relation_data(os_rel_id, "kfp-api-provider", os_data)
+    # Act
+    harness.charm.on.install.emit()
 
-    harness.begin_with_initial_hooks()
+    # Assert
+    assert isinstance(harness.charm.kfp_api_relation.status, BlockedStatus)
 
-    # confirm that we can serialize the pod spec and that the unit is active
-    yaml.safe_dump(harness.get_pod_spec())
-    assert harness.charm.model.unit.status == ActiveStatus()
 
+def test_pebble_services_running(harness):
+    """Test that if the Kubernetes Component is Active, the pebble services successfully start."""
+    # Arrange
+    harness.begin()
+    harness.set_can_connect("persistenceagent", True)
+
+    # Mock:
+    # * leadership_gate_component_item to be active and executed
+    # * kubernetes_resources_component_item to be active and executed
+    # * object_storage_relation_component to be active and executed, and have data that can be
+    #   returned
+    harness.charm.leadership_gate.executed = True
+    harness.charm.leadership_gate.get_status = MagicMock(return_value=ActiveStatus())
+    harness.charm.kfp_api_relation.executed = True
+    harness.charm.kfp_api_relation.component.get_data = MagicMock(return_value=MOCK_KFP_API_DATA)
+
```

Comment:
Same comment as above about the `...executed = True`
```suggestion
    harness.charm.leadership_gate.get_status = MagicMock(return_value=ActiveStatus())
    harness.charm.kfp_api_relation.component.get_data = MagicMock(return_value=MOCK_KFP_API_DATA)

```

Reply:
Done.